packages/core/src/shared/telemetry/activation.ts

```diff
@@ -49,12 +50,26 @@ export async function activate(
     globals.telemetry = await DefaultTelemetryService.create(extensionContext, awsContext, getComputeRegion())
 
     try {
-        globals.telemetry.telemetryEnabled = config.isEnabled()
+        await globals.telemetry.setTelemetryEnabled(config.isEnabled())
 
         extensionContext.subscriptions.push(
-            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(event => {
+            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(async event => {
                 if (event.key === 'telemetry') {
-                    globals.telemetry.telemetryEnabled = config.isEnabled()
+                    const val = config.isEnabled()
+                    await telemetry.aws_modifySetting.run(async () => {
+                        telemetry.record({ settingId: 'telemetryEnabled' })
```

Comment:
Typically the settingId would be the actual setting id, either `amazonQ.telemetry` or `aws.telemetry`